translator/translate/otel/processor/resourceprocessor/translator.go

```diff
@@ -65,6 +69,54 @@ func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
 	c := confmap.NewFromStringMap(map[string]any{
 		"attributes": attributes,
 	})
+	if t.Name() == common.PipelineNameContainerInsightsJmx {
```

Comment:
Consider moving this if to under
```
if strings.HasPrefix(t.Name(), common.PipelineNameJmx) {
		attributes = t.getJMXAttributes(conf)
} else if t.Name() == common.PipelineNameContainerInsightsJmx {
```
and set the `attributes` []any slice instead of replacing the confmap.